src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -426,7 +426,7 @@ private EC2AbstractSlave provisionOndemand(TaskListener listener) throws AmazonC
                             updateRemoteTags(ec2, inst_tags, inst.getInstanceId());
                             break;
                         } catch (AmazonServiceException e) {
-                            if (e.getErrorCode().equals("InvalidInstanceRequestID.NotFound")) {
+                            if (e.getErrorCode().equals("InvalidInstanceID.NotFound")) {
```

Comment:
> If you are unsure about previous ErrorCode

I am not 100% sure, that's why I submitted it for review linking to research I did in the Jenkins bugtracker ticket. If the maintainer asks to leave the old code, I can add it, but I'm reasonably sure it's just a typo, so leaving it in the code would be shame.